ext/js/language/text-scanner.js

```diff
@@ -815,6 +822,15 @@ class TextScanner extends EventDispatcher {
         return entries;
     }
 
+    _getManualSelectionChangeByUserListener(capture) {
```

Comment:
Define callback similar to this:
https://github.com/FooSoft/yomichan/blob/4e7762edacb6cfdb49207bbf43102a04706c78a1/ext/js/language/text-scanner.js#L369-L372
https://github.com/FooSoft/yomichan/blob/4e7762edacb6cfdb49207bbf43102a04706c78a1/ext/js/language/text-scanner.js#L806-L808

Probably just put it after that one, can name it just `_onSelectionChange2` for simplicity. You can also omit the `capture` parameter, as it isn't applicable to `selectionchange`.

Reply:
You can choose a more meaningful name than `_onSelectionChange2`, it was for example and to show that I wouldn't mind if you did name it that. Rationale behind this request is for consistency with the rest of the file (and application in general).

> easier for the JS VM to optimize

I don't think it makes any difference realistically, but a while back I decided on using `.bind` for event handlers rather than arrow functions, so this kind of falls under that decision.

Reply:
You know, I went to implement that change and I tried to pick a good name and the best thing I could come up with was `onSelectionChangeListenerThatSetsWhetherUserHasSelectedAnythingManually` and this would be ridiculous. The thing about event listeners is that I am not sure if they can ever have a specific name. A selection change listener is a selection change listener! I guess this is one of the reasons they rarely get identifiers, and if they do, they are rarely descriptive of the action.

I guess if you think that consistency is key, I can rename this to `_onSelectionChange2`.

P.S. This shouldn't matter much but apparently [binds can even be slower than closures](https://stackoverflow.com/questions/17638305/why-is-bind-slower-than-a-closure)?

Reply:
> I guess if you think that consistency is key, I can rename this to `_onSelectionChange2`.

Yeah, the rationale I've used for putting event handlers directly on the body is because:

1. Sometimes they are non-trivial. I prefer to not have 10s of lines of callback code embedded into functions.
2. Due to 1, for consistency, even short callbacks are set up the same way.
3. Makes the event handlers a bit easier to find all of them. All of the `_on*` functions are generally located in the same area.

And as far as naming goes, I generally follow just `_on<Object (optional)><EventName>`, without trying to include in the name what the code is doing or why. So following that pattern, adding a `2` is the only meaningful thing. I suppose the handlers could also be merged into a single handler which is always attached.

> P.S. This shouldn't matter much but apparently [binds can even be slower than closures](https://stackoverflow.com/questions/17638305/why-is-bind-slower-than-a-closure)?

The question is from 8 years ago so things have potentially changed; I've seen this be the case for a few old topics I've seen on there. It also seems to be referencing that `bind` itself can be slower than assigning a closure. Invocation of the function later on is basically the same.

This was also a design decision that was made a while back in #359.

Reply:
Perhaps naming a selection change listener a selection change listener is fine if you only have one of these. But there was one already! Eugh. Well, I looked around and found the method `_onTouchMovePreventScroll`, which seems to be a friendly cousin of `_onTouchMove`. Now this is not too bad! `TouchMove` describes when it's called, and `PreventScroll` describes what it does. *And* if you look for `_onTouchMove` you'll find both.

I tried phrasing it this way and came up with `_onSelectionChangeCheckUserSelection`? Yes it is still akward, but at least
  * you won't confuse it with `_onSelectionChange`
  * it still fits all lines
  * consistent with the rest of the code
  * you'll find it if you look for `_onSelectionChange`!

How about this? If you don't like it, `_onSelectionChange2` it is.

Reply:
Looks good to me